File: crates/chidori-js/benchmarks/workloads/closures.js
// Closures + higher-order calls in a loop (upvalue capture/read).
// Mirrors the `closures` criterion micro-benchmark, scaled up.
const N = 1_000_000;
function adder(n) {
  return function (x) {
    return x + n;
  };
}
const f = adder(5);
let s = 0;
for (let i = 0; i < N; i++) s = f(s) - 4;
console.log("RESULT=" + s);

File: crates/chidori-js/benchmarks/workloads/closures.py
# Python twin of closures.js — closure capture + higher-order calls in a loop.
N = 1_000_000


def adder(n):
    def add(x):
        return x + n

    return add


f = adder(5)
s = 0
for i in range(N):
    s = f(s) - 4
print("RESULT=" + str(s))
